File: projeto_ed_para_praticar/repository/fila/Enfileiravel.java
package repository.fila;

//Queueable
public interface Enfileiravel {
     //Métodos principais
    void enfileirar(Object dado);           //C - enqueue
    Object frente();                        //R - front
    void atualizarInicio(Object novoDado);  //U - updateFront
    void atualizarFim(Object novoDado);     //U - updateRear
    Object desenfileirar();                 //D - dequeue
    
    //Métodos auxiliares
    boolean estaVazia();                 //isEmpty
    boolean estaCheia();                 //isFull
    String imprimir();                   //print
}

File: fila/Fila Dinâmica/Kotlin/Fila Genérica/Enfileiravel.kt
interface Enfileiravel<T> {
    // Metodos principais
    fun enfileirar(dado: T?)
    fun desenfileirar(): T?
    fun atualizar(dado: T?)    
    fun frente(): T?
    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}